dali/image/tiff_libtiff.cc

```diff
@@ -260,6 +260,14 @@ TiffImage_Libtiff::TiffImage_Libtiff(const uint8_t *encoded_buffer,
     TIFFGetFieldDefaulted(tif_.get(), TIFFTAG_ROWSPERSTRIP, &rows_per_strip_));
   LIBTIFF_CALL(
     TIFFGetFieldDefaulted(tif_.get(), TIFFTAG_COMPRESSION, &compression_));
+
+  uint16_t photometric_interpretation;
+  LIBTIFF_CALL(
+    TIFFGetFieldDefaulted(tif_.get(), TIFFTAG_PHOTOMETRIC, &photometric_interpretation));
```

Comment:
```suggestion
  TIFFGetFieldDefaulted(tif_.get(), TIFFTAG_PHOTOMETRIC, &photometric_interpretation);
```
Apparently libtiff returns 1 if the tag was not there, even though it's filling the default value.

Reply:
Thank you! I dug a bit deeper and indeed, that's what the [doc](http://www.libtiff.org/man/TIFFGetField.3t.html) for `TIFFGetField`/`TIFFGetFieldDefaulted` says:

> 1 is returned if the tag is defined in the current directory; otherwise a 0 is returned.

But this parser already has like 4 `TIFFGetFieldDefaulted`s wrapped in `LIBTIFF_CALL`! How did it even work before? It turns out (see the [source here](https://gitlab.com/libtiff/libtiff/-/blob/master/libtiff/tif_aux.c#L208-337)) that if the tag was **not** defined in a directory, but **has** a default value, it still returns 1.

The problem with `TIFFTAG_PHOTOMETRIC` is that it does not have any default value defined and that's why `TIFFGetFieldDefaulted` fails.

Reply:
So I'll fix that in a moment and re-run the CI

Reply:
Fixed, respinning the CI...

Reply:
Perhaps we should remove LIBTIFF_CALL from all calls where we use TIFFGetFieldDefaulted

Reply:
`LIBTIFF_CALL` around `TIFFGetFieldDefaulted` is fine as long as the tag has some default value defined in libtiff. It was my mistake to use `TIFFGetFieldDefaulted` for getting `TIFFTAG_PHOTOMETRIC` which has no default value. The other usages are correct so I think they can stay as they are

Reply:
Understood. Thanks for the explanation